interp/src.go

```diff
@@ -128,32 +128,33 @@ func (interp *Interpreter) importSrc(rPath, importPath string) (string, error) {
 	// Register source package in the interpreter. The package contains only
 	// the global symbols in the package scope.
 	interp.mutex.Lock()
-	interp.srcPkg[importPath] = interp.scopes[importPath].sym
+	gs := interp.scopes[importPath]
+	interp.srcPkg[importPath] = gs.sym
 	interp.pkgNames[importPath] = pkgName
 
 	interp.frame.mutex.Lock()
 	interp.resizeFrame()
 	interp.frame.mutex.Unlock()
 	interp.mutex.Unlock()
 
-	// Once all package sources have been parsed, execute entry points then init functions
+	// Once all package sources have been parsed, execute entry points then init functions.
 	for _, n := range rootNodes {
 		if err = genRun(n); err != nil {
 			return "", err
 		}
 		interp.run(n, nil)
 	}
 
-	// Wire and execute global vars
-	n, err := genGlobalVars(rootNodes, interp.scopes[importPath])
+	// Wire and execute global vars in global scope gs.
+	n, err := genGlobalVars(rootNodes, gs)
 	if err != nil {
 		return "", err
 	}
 	interp.run(n, nil)
 
-	// Add main to list of functions to run, after all inits
-	if m := interp.main(); m != nil {
-		initNodes = append(initNodes, m)
+	// Add main to list of functions to run, after all inits.
+	if m := gs.sym[mainID]; m != nil && skipTest {
```

Comment:
Same remark as above. I'm not sure this is correct, as importPath may be something other than "main".

Reply:
will replace L156 by:
```go
if m := gs.sym[mainID]; pkgName == mainID && m != nil && skipTest {
```